src/AppInstallerCommonCore/Public/winget/MsixManifestValidation.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "AppInstallerErrors.h"
+#include "AppInstallerMsixInfo.h"
+#include "winget/Manifest.h"
+#include "winget/ManifestValidation.h"
+
+namespace AppInstaller::Manifest
+{
+    struct MsixManifestValidation
+    {
+        MsixManifestValidation(ValidationError::Level validationErrorLevel) : m_validationErrorLevel(validationErrorLevel) {}
+
+        std::vector<ValidationError> Validate(
+            const Manifest &manifest,
+            const ManifestInstaller &installer);
+    private:
+        std::map<std::string, std::shared_ptr<Msix::MsixInfo>> m_msixInfoCache;
+        ValidationError::Level m_validationErrorLevel;
+
+        // Get Msix info from installer url, or load it from cache. Return null
+        // pointer if failed to process installer url.
+        std::shared_ptr<Msix::MsixInfo> GetMsixInfo(
+            std::string installerUrl,
+            std::vector<ValidationError>& errors);
+
+        // Get manifest installer minimum OS version or nullopt if failed to
+        // parse input.
+        std::optional<Msix::OSVersion> GetManifestInstallerMinOSVersion(
+            std::string minOSVersion,
+            std::vector<ValidationError>& errors);
+
+        // Validate Msix package family name.
+        void ValidateMsixManifestPackageFamilyName(
+            Utility::NormalizedString msixPackageFamilyName,
+            std::string manifestPackageFamilyName,
+            std::vector<ValidationError>& errors);
```

Comment:
Should both input here be of the same type: `std::string` or `NormalizedString`?
- First input `msixPackageFamilyName` is the value from the MSIX package.
- Second input `manifestPackageFamilyName` is the value from the YAML manifest.

Reply:
`NormalizedString`, which is what the manifest already contains I believe.

Reply:
You're right, I misread `string_t` (which is `= Utility::NormalizedString`) as `std::string`